### PR TITLE
ci: add jjbb jobs and utility script

### PR DESCRIPTION
**Pull request by jonahbull**
Add the definition for the pull request job so that it can be managed from this repo. The values in defaults.yml will be merged with the pull request definition. `check_paths_for_matches` is a utility script used to prevent unnecessary builds from happening on every push to master due to problems with the Jenkins `git` plugin.

### CODE REVIEW

**Comment by jonahbull** [comment]
This is part of the project to move JJB job definitions into the product repos themselves so they can be updated there without having to open PRs against the Infra repo.

cc: @spalger

File: .ci/jobs/defaults.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    logrotate:
+      daysToKeep: 30
+      numToKeep: 100
+    properties:
+    - github:
+        url: https://github.com/elastic/elastic-charts/
+    - inject:
+        properties-content: 'HOME=$JENKINS_HOME
+
+          '
```
Comment (markov00) [inline .ci/jobs/defaults.yml]: Seems to be something wrong or missing here.

Comment (jonahbull) [reply .ci/jobs/defaults.yml]: I believe this is as intended, JJB has weird handling of newlines/formatting.

**Comment by jonahbull** [comment]
@markov00 That's right, if you want to add jobs you can just drop them in the `.ci/jobs` directory.

**Comment by jonahbull** [comment]
@markov00 If this looks okay I'd like to get it merged so I can wire up the jobs on the Infra side. Thanks!

**Pull request merged**